code/modules/mob/living/living.dm

```diff
@@ -749,19 +749,25 @@
 // Override if a certain mob should be behave differently when placing items (can't, for example)
 /mob/living/stripPanelEquip(obj/item/what, mob/who, where, var/silent = 0)
 	what = get_active_hand()
-	if(what && (what.flags & NODROP))
-		to_chat(src, "<span class='warning'>You can't put \the [what.name] on [who], it's stuck to your hand!</span>")
-		return
 	if(what)
+		if(what.flags & NODROP)
+			to_chat(src, "<span class='warning'>You can't put \the [what.name] on [who], it's stuck to your hand!</span>")
+			return
 		if(!what.mob_can_equip(who, where, 1))
+			if((where == slot_l_hand || where == slot_r_hand) && isliving(who))
+				var/mob/living/L = who
+				if(L.resting)
+					to_chat(src, "<span class='warning'>[who.name] can't hold \the [what.name] while resting.</span>")
+					return
 			to_chat(src, "<span class='warning'>\The [what.name] doesn't fit in that place!</span>")
 			return
 		if(!silent)
 			visible_message("<span class='notice'>[src] tries to put [what] on [who].</span>")
 		if(do_mob(src, who, what.put_on_delay))
-			if(what && Adjacent(who))
-				unEquip(what)
-				who.equip_to_slot_if_possible(what, where, 0, 1)
+			if(what && Adjacent(who) && !(what.flags & NODROP))
+				unEquip(what)	
+				if(!who.equip_to_slot_if_possible(what, where, 0, 1))
+					what.loc = who.loc
```

Comment:
What are you doing?

Reply:
If the person is lying down on the floor the check above will fail. Making the object go to the floor where the person is lying down at.

Reply:
`unEquip` will drop the item at your feet, so this check is kinda...redundant.

Also, unless there's an exceptionally good reason to do so, don't set `loc` directly. We use `forceMove` for nearly everything now because of lights and ensuring stuff liked `Crossed` gets called.

Reply:
Copy that will keep that in mind. I'll just let it drop at the owners feet then. Thanks

Reply:
our inventory code is pretty dated, but ideally, there would be a proc that would unequip an item and equip it to a slot only if something like `equip_to_slot_if_possible` returns `TRUE`.

I'm pretty sure TG has this now, but.....

Reply:
the whole refactoring of the inventory will be for another PR. I found put_in_hands() which will suit the needs a bit. Combined with unEquip